config/v1/types_insights.go

```diff
@@ -0,0 +1,72 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// Insights holds cluster-wide information about Insights.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type Insights struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec InsightsSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status InsightsStatus `json:"status"`
+}
+
+type InsightsSpec struct {
+	// GatherConfig spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	// +optional
+	GatherConfig *GatherConfig `json:"gatherConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+}
+
+type GatherConfig struct {
+	// DataPolicy allows user to enable additional global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	// +kubebuilder:default=NoPolicy
+	DataPolicy DataPolicy `json:"dataPolicy"`
+	// ForceGatherReason enables user to force Insights data gathering by setting a new reason.
+	// When there is some gathering in the progress then it is interrupted.
+	// When all the gatherers are deactivated by the `DisabledGatherers`, nothing happens.
+	// When the forced gathering is finished then the value is cleared.
+	ForceGatherReason string `json:"forceGatherReason"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can be disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	DisabledGatherers []string `json:"disabledGatherers"`
+}
+
+const (
+	// No data obfuscation
+	NoPolicy DataPolicy = "NoPolicy"
+	// IP addresses and cluster domain name is obfuscated
+	IPsAndClusterDomainPolicy DataPolicy = "IPsAndClusterDomainPolicy"
```

Comment:
I agree that we don't need to repeat the word policy, but I would do it as `None` and `IPsAndClusterDomain`.... `ObfuscateNetworking` sounds too general to me

Reply:
The only thing I would say is that `IPsAndClusterDomain` doesn't tell me what it's doing with those pieces of information, I think there should be some description of that within the policy name

Reply:
Fair point :) What about to call the attribute `ObfuscationPolicy` rather than `DataPolicy`? Just for the context - this was suggested by David in https://github.com/openshift/enhancements/pull/1037#discussion_r847662437

Reply:
This is not about the `DataPolicy` per se, it's about not using that as the value, ie:
```golang
IPsAndClusterDataPolicy DataPolicy = "IPsAndCluster"
```
will match what we have elsewhere.

Reply:
Ahh OK. Got it. Thanks.

Reply:
Updated